draft-ietf-rats-corim.md

```diff
@@ -2070,133 +2042,195 @@ Logically, new ECT entries are appended to the existing ACS.
 
 Profile writers SHOULD use CBOR tags for widely applicable comparison methods to ease Verifier implementation compliance across profiles.
 
-The following subsections define the comparison algorithms for the `measurement-values-map` keys defined by this specification.
+The following subsections define the comparison algorithms for the `measurement-values-map` codepoints defined by this specification.
 
 ##### Comparison for svn entries
 
-The value stored under `measurement-values-map` key 1 is an SVN, which must
-have type UINT.
+The value stored under `measurement-values-map` key 1 is an SVN, which must have type `uint`.
 
-If the Reference value for `measurement-values-map` key 1 is an untagged UINT or
-a UINT tagged with #6.552 then an equality comparison is performed. If the value
-of the SVN in ACS is not equal to the value in the Reference
-Value then the Reference Value does not match.
+If the condition ECT value for `measurement-values-map` key 1 is an untagged `uint` or a `uint` tagged with #6.552 then an equality comparison is performed.
+The comparison MUST return true if the value of the SVN in the candidate entry is equal to the value in the condition ECT.
 
-If the Reference value for `measurement-values-map` key 1 is a UINT tagged with
-#6.553 then a minimum comparison is performed. If the value of the SVN in
-ACS less than the value in the Reference Value then the
-Reference Value does not match.
+If the condition ECT value for `measurement-values-map` key 1 is a `uint` tagged with #6.553 then a minimum comparison is performed.
+The comparison MUST return true if the value of the SVN in the candidate entry is less than the value in the condition ECT.
 
 ##### Comparison for digests entries {#sec-cmp-digests}
 
-The value stored under `measurement-values-map` key 2,
-or a value tagged with
-#6.TBD is a digest entry.
-It contains one or more digests, each measuring the
-same object. A Reference Value may contain multiple digests, each with a
-different algorithm acceptable to the Reference Value provider. If the
-digest in Evidence contains a single value with an algorithm and value
-matching one of the algorithms and values in the Reference Value then it
-matches.
+The value stored under `measurement-values-map` key 2 is a digest entry.
+It contains one or more digests, each measuring the same object.
+When multiple digests are provided, each represents a different algorithm acceptable to the condition ECT author.
+
+In the simple case, a condition ECT digests entry containing one digest matches matches a candidate entry containing a single entry with the same algorithm and value.
+
+To prevent downgrade attacks, if there are multiple algorithms in common between the condition ECT and candidate entry then the digests calculated using all common algorithms must match.
```

Comment:
```suggestion
To prevent downgrade attacks, if there are multiple algorithms in common between the condition ECT and candidate entry, then the bytes paired with common algorithms must be equal.
An `alg` matches another if they correspond to the same hash algorithm in the [IANA.named-information] registry.
A Verifier SHOULD fail the comparison if a condition `alg` does not have a corresponding entry in the IANA registry.
```

Reply:
@andrew-draper to correct digest comparison to match support both of text and integer for hash alg

Reply:
I have been thinking about what we discussed in the call and I think there may be a security issue with it. As I remember, the decision in the call was that a verifier should treat the string and integer names for an algorithm as identical.
This leads to an issue if the registry is every extended, as some verifiers will not understand the new algorithms. If a condition ECT uses the numeric version and an ACS entry uses the string version then a downgrade attack is possible.
I am proposing text below which recommends that the integer version be used when it is available, and does not mandate comparing strings and integers.